stdlib/nats/src/main/java/org/ballerinalang/nats/nativeimpl/Utils.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.nats.nativeimpl;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BLangVMErrors;
+import org.ballerinalang.connector.api.BLangConnectorSPIUtil;
+import org.ballerinalang.connector.api.Resource;
+import org.ballerinalang.connector.api.Service;
+import org.ballerinalang.model.types.BTypes;
+import org.ballerinalang.model.values.BError;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utilities for producing and consuming via NATS sever.
+ */
+public class Utils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Utils.class);
+
+    /**
+     * Message which will be propagated.
+     */
+    private static final String MESSAGE = "message";
+
+    /**
+     * Constructs a BallerinaException and throw it.
+     *
+     * @param message   error message which should be included.
+     * @param context   context of the ballerina call.
+     * @param throwable the exception stacktrace.
+     */
+    public static void throwBallerinaException(String message, Context context, Throwable throwable) {
+        LOGGER.error(message, throwable);
+        throw new BallerinaException(message + " " + throwable.getMessage(), throwable, context);
+    }
+
+    /**
+     * Retrieve the receiver object.
+     *
+     * @param context Ballerina context which should be received.
+     * @return the receiver object struct.
+     */
+    public static BMap<String, BValue> getReceiverObject(Context context) {
+        return (BMap<String, BValue>) context.getRefArgument(0);
+    }
+
+    /**
+     * Creates an error message.
+     *
+     * @param context context which is invoked.
+     * @param errCode the error code.
+     * @param errMsg  the cause for the error.
+     * @return an error which will be propagated to ballerina user.
+     */
+    public static BError createError(Context context, String errCode, String errMsg) {
```

Comment:
There's an API for creating errors in `BLangVMErrors`. Why not use that?

Reply:
This method uses BLangVMErrors :) 

What it does is to include the error message to the IOError record and create an error through BLangVMErrors. We call this in several places hence introduced this as a util method